ext/tags.go

```diff
@@ -88,6 +88,18 @@ var (
 
 	// Error indicates that operation represented by the span resulted in an error.
 	Error = boolTagName("error")
+
+	//////////////////////////////////////////////////////////////////////
+	// PerfEvent Tag
+	//////////////////////////////////////////////////////////////////////
+
+	// PerfEvent records the list of events given in a comma separated string (e.g.
+	// "cpu-cycles,cache-misses,branch-misses") for a span.
+	// The tracer is expected to start the events in PerfEvent using the "perfevents"
+	// package (https://github.com/opentracing-contrib/perfevents/go) . On span
+	// Finish(), the events must be closed.
+
+	PerfEvent = stringTagName("perfevents")
```

Comment:
I see, your use cases is slightly different from what I created the observer for, i.e. I used it for emitting metrics, something that's triggered by span events but does not have to come back to span as data. I think if the OnStartSpan() method would pass the actual span to the observer, then you can store it in the SpanObserver created for that span (circular ref - hope go can manage) and write to that span from OnFinish().